src/test/java/teammates/test/cases/ui/StudentProfilePictureActionTest.java

```diff
@@ -189,22 +190,104 @@ private void testActionForStudentWithEmptyGoogleId() {
+    protected void testActionWithEmailAndCourseUnauthorisedInstructorOrStudent() {
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, StringHelper.encrypt(student.email),
+                Const.ParamsNames.COURSE_ID, StringHelper.encrypt(student.course)
+        };
+        
         ______TS("Failure case: instructor not from same course");
         AccountAttributes unauthInstructor = dataBundle.accounts.get("instructor1OfCourse2");
         gaeSimulation.loginAsInstructor(unauthInstructor.googleId);
 
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("User is not in the course that student belongs to", uae.getMessage());
+        }
+        
+        ______TS("Failure case: student not from same course");
+        AccountAttributes unauthStudent = dataBundle.accounts.get("student1InArchivedCourse");
+        gaeSimulation.loginAsStudent(unauthStudent.googleId);
+
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("User is not in the course that student belongs to", uae.getMessage());
+        }
+        
+        ______TS("Failure case: student not from same team");
+        
+        StudentAttributes studentFromDifferentTeam = dataBundle.students.get("student5InCourse1");
+        gaeSimulation.loginAsStudent(studentFromDifferentTeam.googleId);
+
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("Student does not have enough privileges to view the photo", uae.getMessage());
+        }
+    }
+    
+    protected void testActionWithEmailAndCourseUnauthorisedInstructorOrStudentMasquerade() {
         String[] submissionParams = new String[] {
                 Const.ParamsNames.STUDENT_EMAIL, StringHelper.encrypt(student.email),
                 Const.ParamsNames.COURSE_ID, StringHelper.encrypt(student.course)
         };
-
-        action = getAction(submissionParams);
+        
+        ______TS("Failure case: unauthorised student masqueraded as a student from same team");
+        AccountAttributes unauthStudent = dataBundle.accounts.get("student1InArchivedCourse");
+        gaeSimulation.loginAsStudent(unauthStudent.googleId);
+        try {
+            action = getAction(addUserIdToParams(student.googleId, submissionParams));
+            signalFailureToDetectException();
+        } catch (UnauthorizedAccessException uae) {
+            ignoreExpectedException();
```

Comment:
Is there an error message that can be used for assertion here? Similarly for the one after this.